packages/lib/hooks/usePagination.ts

```diff
@@ -0,0 +1,28 @@
+import type { PaginationState } from "@tanstack/react-table";
+import { useState, useMemo } from "react";
+
+export function usePagination({
+  defaultPageIndex = 1,
+  defaultPageSize = 20,
+}: {
+  defaultPageIndex?: number;
+  defaultPageSize?: number;
+}) {
+  const [{ pageIndex, pageSize }, setPagination] = useState<PaginationState>({
+    pageIndex: defaultPageIndex,
+    pageSize: defaultPageSize,
+  });
+
+  const pagination = useMemo(
+    () => ({
+      pageIndex,
+      pageSize,
+    }),
+    [pageIndex, pageSize]
+  );
+
+  return {
+    pagination,
+    setPagination,
+  };
+}
```

Comment:
This isnt actually being used - but its a useful hook if we ever need skip based pagination.  (some of our tables dont support cursor based that well)